technical-reports/format/composite-types.md

```diff
@@ -439,7 +440,7 @@ Represents a typographic style. The `$type` property MUST be set to the string `
         "fontSize": "{font.size.smallest}",
         "fontWeight": "{font.weight.normal}",
         "letterSpacing": "0px",
-        "lineHeight": "1"
+        "lineHeight": "24px"
```

Comment:
The type of `lineHeight` is ambiguous if both `1.2` and `"24px"` are allowed.

It is no longer possible to determine the value based on the current specification.
The specification only describes the `$type` property as a source for type information.

It is not possible to allow multiple types on the same field in a composite token.

Reply:
I think it's a bit of a grey area. The spec just says that "names and types of sub-values are pre-defined". That doesn't necessarily _forbid_ a composite type for specifying a range of permitted types for one of its sub-values.

However, I agree that this probably needs to be made more explicit one way or the other. I only updated the `typography` type in this PR because `string` is now gone. To avoid this PR getting side-tracked on updates to the `typography` type (which is not this PR's purpose), I can perhaps change it to only accept `dimension` values for now. That's hopefully a more "minimal" change that achieves the original aim of not replacing the `string` type with something that's still valid.

Reply:
> I think it's a bit of a grey area. The spec just says that "names and types of sub-values are pre-defined". That doesn't necessarily forbid a composite type for specifying a range of permitted types for one of its sub-values.

This is definitely not a grey area :)
The specification doesn't define how a token parser must handle a composite token with fields that can have one of multiple types. Because it is not defined it can not be implemented. That it can not be implemented means that it isn't allowed/supported.

I think this issue is relevant : https://github.com/design-tokens/community-group/issues/199

-----

> I can perhaps change it to only accept dimension values for now.

Isn't `number` a better fit?
All the example previously were numbers but "embedded" in a JSON string.

I am unsure if this was something that leaked from CSS or not as it is typical for CSS that line height is a fraction of font size and therefor not a dimension.

Reply:
I've changed it to just `number` for now. Like you said, it's closer to what we had previously.

Reply:
Thank you for these changes 🙇

Reply:
Thanks for your review comments :-)